ayushma/views/chat.py

```diff
@@ -0,0 +1,27 @@
+import openai
+from django.conf import settings
+from drf_spectacular.utils import extend_schema
+from rest_framework import authentication, permissions
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from ..utils.openaiapi import get_embedding
+
+
+class TextToVector(APIView):
+    """
+    View to convert text to vector embeddings using OpenAI's API
+    """
+
+    authentication_classes = [authentication.TokenAuthentication]
+    permission_classes = [permissions.AllowAny]
+
+    @extend_schema(
+        tags=("chat",),
+        operation_id="text-to-vector",
+    )
+    def post(self, request):
+        text = request.data.get("text")
+        openai.api_key = settings.OPENAI_API_KEY
+        embedding = get_embedding(text=text)
```

Comment:
Maximum token size that this endpoint can receive is 8192, we can check the input token size using tiktoken function and split the text or send just the text within limit, so that user can get answer 🤔

Reply:
Yep, we need to split the text for huge tokens @siddnikh

Reply:
@ishanExtreme @skks1212 Should we take this up separately in https://github.com/coronasafe/ayushma/issues/17 ?